internal/csi/storage/modelregistry_provider.go

```diff
@@ -99,16 +100,19 @@ func (p *ModelRegistryProvider) DownloadModel(modelDir string, modelName string,
 		return fmt.Errorf("%w %s", ErrModelArtifactEmptyURI, *modelArtifact.Id)
 	}
 
+	log.Printf("Extracting protocol from model artifact URI: %s", apiutils.SafeString(modelArtifact.Uri))
 	protocol, err := p.extractProtocol(*modelArtifact.Uri)
 	if err != nil {
 		return err
 	}
 
+	log.Printf("Getting KServe provider for protocol: %s", protocol)
 	provider, err := kserve.GetProvider(p.Providers, protocol)
 	if err != nil {
 		return err
 	}
 
+	log.Printf("Delegating to KServe provider to download model with: modelDir=%s, storageUri=%s", modelDir, apiutils.SafeString(modelArtifact.Uri))
```

Comment:
<img width="1230" height="73" alt="image" src="https://github.com/user-attachments/assets/2350c447-a65d-490d-8cf7-f4536f22d79e" />

please note KServe does not seem to offer any helpful logging ootb here :( 
but at least we can log everything the Model Registry CSI code determined before delegating to KServe